File: objective-c/PersonTests/PersonTest.m
//
//  PersonTest.m
//  Person
//
//  Created by Ray Zane on 8/7/15.
//  Copyright (c) 2015 Ray Zane. All rights reserved.
//

#import <Cocoa/Cocoa.h>
#import <XCTest/XCTest.h>
#import "Person.h"

@interface PersonTest : XCTestCase {
    Person *jeff;
    Person *susy;
}

@end

@implementation PersonTest

- (void)setUp {
    [super setUp];

    jeff = [[Person alloc] initWithName:@"Jeff"];
    susy = [[Person alloc] initWithName:@"Susy"];

}

- (void)testName {
    XCTAssert([jeff.name isEqualToString:@"Jeff"]);
}

- (void)testGreetHelloJeff {
    NSString *greeting = [jeff greet:@"Hello"];
    XCTAssert([greeting isEqualToString:@"Hi, I'm Jeff"]);
}

- (void)testGreetHelloSusy {
    NSString *greeting = [susy greet:@"Hello"];
    XCTAssert([greeting isEqualToString:@"Hi, I'm Susy"]);
}

- (void)testGreetGoodbye {
    NSString *greeting = [jeff greet:@"Goodbye"];
    XCTAssert([greeting isEqualToString:@"Bye"]);
}

- (void)testGreetOther {
    NSString *greeting = [jeff greet:@"Yo"];
    XCTAssert([greeting isEqualToString:@"Yo"]);
}

@end

File: csharp/Person/PersonTest.cs
using NUnit.Framework;
using System;

namespace Person
{
	[TestFixture]
	public class PersonTest
	{

		private Person jeff;
		private Person susy;

		[SetUp]
		public void Setup ()
		{
			jeff = new Person ("Jeff");
			susy = new Person ("Susy");
		}

		[Test]
		public void TestName ()
		{
			Assert.AreEqual (jeff.Name, "Jeff");
		}

		[Test]
		public void TestGreetHelloJeff ()
		{
			Assert.AreEqual ("Hi, I'm Jeff", jeff.Greet ("Hello"));
		}

		[Test]
		public void TestGreetHelloSusy ()
		{
			Assert.AreEqual ("Hi, I'm Susy", susy.Greet ("Hello"));
		}

		[Test]
		public void TestGreetGoodbye ()
		{
			Assert.AreEqual ("Bye", jeff.Greet ("Goodbye"));
		}

		[Test]
		public void TestGreetOther ()
		{
			Assert.AreEqual ("Yo", jeff.Greet ("Yo"));
		}
	}
}
